zerver/tests/test_event_system.py

```diff
@@ -1157,47 +1157,47 @@ def test_queries(self) -> None:
 
         self.login_user(user)
 
-        with self.assert_database_query_count(43):
+        with self.assert_database_query_count(45):
             with mock.patch("zerver.lib.events.always_want") as want_mock:
                 fetch_initial_state_data(user)
 
         expected_counts = dict(
-            alert_words=1,
-            custom_profile_fields=1,
-            default_streams=1,
-            default_stream_groups=1,
-            drafts=1,
-            message=1,
-            muted_topics=1,
-            muted_users=1,
-            onboarding_steps=1,
-            presence=1,
-            realm=1,
-            realm_bot=1,
-            realm_domains=1,
-            realm_embedded_bots=0,
-            realm_incoming_webhook_bots=0,
-            realm_emoji=1,
-            realm_filters=0,
-            realm_linkifiers=0,
-            realm_playgrounds=1,
-            realm_user=4,
-            realm_user_groups=5,
-            realm_user_settings_defaults=1,
-            recent_private_conversations=1,
-            scheduled_messages=1,
-            starred_messages=1,
-            stream=3,
-            stop_words=0,
-            subscription=4,
-            update_display_settings=0,
-            update_global_notifications=0,
-            update_message_flags=5,
-            user_settings=0,
-            user_status=1,
-            user_topic=1,
-            video_calls=0,
-            giphy=0,
+            alert_words=2,
+            custom_profile_fields=2,
+            default_streams=2,
+            default_stream_groups=2,
+            drafts=2,
+            message=2,
+            muted_topics=2,
+            muted_users=2,
+            onboarding_steps=2,
+            presence=2,
+            realm=4,
```

Comment:
Previously the query count for `realm` was `1`. While it should have been 3 but it was 1 because the related fields like `new_stream_announcements_stream` and signup_announcements_stream` were already fetched before when testing for all event types, so the queries to fetch them were not made again. I tested this by testing the query count for `realm` event type in a separate test.

And it is 4 now due to one more query for refetching realm.

Reply:
And for all other event types as well one query is added for refetching realm.

Reply:
Oh, shoot, this is annoying; the point of this test is to show how many marginal queries each of these things requires. I'm not sure what the best way to fix this is; but one option would be to move this refetch further up the stack into `do_events_register`, such that these lower-level tests that are just looking at `fetch_initial_state_data` can start with the refetched `realm` object? Might require some plumbing to have `realm` be a required parameter, rather than optional.